### ISSUE TITLE
Bug: False positive from DL3040 triggered by "rm -rf" of any folder before "dnf install"

### ISSUE THREAD

**Issue by Wuodan** [hadolint/hadolint#1250]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched for existing issues before opening this one

### Expected behavior

**DL3040 "`dnf clean all` missing after dnf command"** does not fire for
```
RUN rm -rf /foo && \
    dnf install -y bash && \
    dnf clean all
```

### Actual behavior

DL3040 is triggered because `rm -rf` is before `dnf install -y`.

### Reason for actual behavior

`Shell.cmdHasArgs` matches when **one** of the expected names is present, **not all** of them.

And DL3040 has an alternative to `dnf clean all`:
```haskell
`Shell.cmdHasArgs "rm" ["-rf", "/var/cache/libdnf5"]`
```
which matches any `rm -rf` and then cleanup is treated as being before `dnf install`.

### Steps to reproduce the behavior

Run `hadolint Dockerfile` with the Dockerfile from below.

and it will report:

```
Dockerfile:4 DL3040 warning: `dnf clean all` missing after dnf command.
``````

**Output of `hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.15.1
```

**Dockerfile**

```Dockerfile
# fedora:46
FROM fedora@sha256:16fd7b21d98a67675452902cf5c2629aad746b66455f9b13c5af4b6d9799add9

RUN rm -rf /foo/bar && \
dnf install -y bash-5.3.15-2.fc45.x86_64 && \
dnf clean all
```

**Additional environment details (OS, stack version, etc.)**

- OS: Fedora (Nobara)
- Originally found when running `megalinter` GitHub action in a GH workflow:
  - Dockerfile: https://github.com/aicage/aicage-custom-samples/blob/main/tests/extensions/Dockerfile
  - GitHub workflow: https://github.com/aicage/aicage-custom-samples/blob/main/.github/workflows/ci_lint.yml
